litestar/middleware/compression/zstd_facade.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Literal
+
+import zstandard as zstd
+
+from litestar.enums import CompressionEncoding
+from litestar.middleware.compression.facade import CompressionFacade
+
+if TYPE_CHECKING:
+    from io import BytesIO
+
+    from litestar.config.compression import CompressionConfig
+
+
+class ZstdCompression(CompressionFacade):
+    __slots__ = ("buffer", "cctx", "compression_encoding", "compressor")
+
+    encoding = CompressionEncoding("zstd")
+
+    def __init__(self, buffer: BytesIO, compression_encoding: Literal["zstd"] | str, config: CompressionConfig) -> None:
+        self.buffer = buffer
+        self.compression_encoding = compression_encoding
+        self.cctx = zstd.ZstdCompressor(level=config.zstd_compress_level)
+        self.compressor = self.cctx.stream_writer(buffer)
+
+    def write(self, body: bytes, final: bool = False) -> None:
```

Comment:
`body` is allowed to be `bytearray` as well: https://github.com/indygreg/python-zstandard/blob/32a31f889e9f968320b76420e8cfd051beec35fc/zstandard/__init__.pyi#L258